src/main.rs

```diff
@@ -366,4 +366,63 @@ mod tests {
             Ok(())
         });
     }
+
+    #[test]
```

Comment:
You're missing the conditional bit here for the new feature.

Reply:
Why? It's in the code itself (line 385)